### ISSUE TITLE
Logging Metrics

### ISSUE THREAD

**Issue by padwasabimasala** [consulate/consulate#12]
Any thoughts on how to go about logging metrics?

It would be nice if consulate logged counts of authentication attempts by type, successes and failures, etc.

Is there any easy way to do that already? Or does it require a change to Consulate?

**Comment by camshaft** [comment]
yeah it wouldn't be hard. we could probably get a lot of it here https://github.com/consulate/consulate/blob/master/lib/application.js#L154

**Comment by padwasabimasala** [comment]
What were you thinking? Wrap the callbacks with logging statements as their defined? Is there a list of available callbacks?
